packages/editor/src/components/Editor2Container.tsx

```diff
@@ -0,0 +1,447 @@
+          style={{
+            // default values will be overridden by theme
+            fontFamily: 'Lato',
+            fontSize: '12px',
+            color: 'white',
+            padding: '16px'
+          }}
+        >
+          {`Scene Loading... ${progress}% - ${resourcePendingQuery.length} assets left`}
+        </div>
+        <CircularProgress />
+      </div>
+    </div>
+  )
+}
+
+/**
+ * Scene Event Handlers
+ */
+
+const onEditorError = (error) => {
+  logger.error(error)
+  if (error['aborted']) {
+    DialogState.setDialog(null)
+    return
+  }
+
+  DialogState.setDialog(
+    <ErrorDialog
+      title={error.title || t('editor:error')}
+      message={error.message || t('editor:errorMsg')}
+      error={error}
+    />
+  )
+}
+
+const onCloseProject = () => {
+  const editorState = getMutableState(EditorState)
+  getMutableState(GLTFModifiedState).set({})
+  editorState.projectName.set(null)
+  editorState.scenePath.set(null)
+  editorState.sceneName.set(null)
+  RouterState.navigate('/studio')
+
+  const parsed = new URL(window.location.href)
+  const query = parsed.searchParams
+
+  query.delete('project')
+  query.delete('scenePath')
+
+  parsed.search = query.toString()
+  if (typeof history.pushState !== 'undefined') {
+    window.history.replaceState({}, '', parsed.toString())
+  }
+}
+
+const onSaveAs = async () => {
+  const { sceneAssetID, projectName, sceneName, rootEntity } = getState(EditorState)
+  const sceneModified = EditorState.isModified()
+  const abortController = new AbortController()
+  try {
+    if (sceneName || sceneModified) {
+      const result: { name: string } | void = await new Promise((resolve) => {
+        DialogState.setDialog(<SaveNewSceneDialog initialName={'New Scene'} onConfirm={resolve} onCancel={resolve} />)
+      })
+      DialogState.setDialog(null)
+      if (result?.name && projectName) {
+        await saveSceneGLTF(sceneAssetID, projectName, result.name, abortController.signal)
+
+        const sourceID = getComponent(rootEntity, SourceComponent)
+        getMutableState(GLTFModifiedState)[sourceID].set(none)
+
+        const newSceneData = await Engine.instance.api
+          .service(assetPath)
+          .find({ query: { assetURL: getState(EditorState).scenePath! } })
+        getMutableState(EditorState).scenePath.set(newSceneData.data[0].assetURL as any)
+      }
+    }
+  } catch (error) {
+    logger.error(error)
+    DialogState.setDialog(
+      <ErrorDialog title={t('editor:savingError')} message={error?.message || t('editor:savingErrorMsg')} />
+    )
+  }
+}
+
+const onImportSettings = () => {
+  DialogState.setDialog(<ImportSettingsPanel />)
+}
+
+const onImportAsset = async () => {
+  const { projectName } = getState(EditorState)
+
+  if (projectName) {
+    try {
+      await inputFileWithAddToScene({ projectName })
+    } catch (err) {
+      NotificationService.dispatchNotify(err.message, { variant: 'error' })
+    }
+  }
+}
+
+const onSaveScene = async () => {
```

Comment:
let's export all these functions from the original file so we arent duplicating them

Reply:
done, we will eventually delete the old route so they will eventually converge to the same file